pybossa/util.py

```diff
@@ -1157,47 +1157,51 @@ def generate_bsso_account_notification(user):
 
 def is_annex_response(key, value):
     """
-    Check whether a response is a odfoa response.
+    Recursively check whether a response is a odfoa response. Returns the check
+    result and odfoa response
     """
-    if key == "odfoa":
-        return True
-
     odfoa_keys = {"version", "source-uri", "odf", "oa"}
-    return all(odfoa_key in value for odfoa_key in odfoa_keys)
+    if all(odfoa_key in value for odfoa_key in odfoa_keys):
```

Comment:
perhaps we need to check type(value) is dict here, in case that value is array or set of `["version", "source-uri", "odf", "oa"]`